module-services/service-cellular/ServiceCellular.cpp

```diff
@@ -1468,17 +1468,16 @@ std::string ServiceCellular::GetScanMode(void)
     return {};
 }
 
-bool ServiceCellular::transmitDtmfTone(uint32_t digit)
+bool ServiceCellular::transmitDtmfTone(DTMFCode code)
 {
     auto channel = cmux->get(CellularMux::Channel::Commands);
     at::Result resp;
     if (channel) {
         auto command           = at::factory(at::AT::QLDTMF);
-        std::string dtmfString = "\"" + utils::singleDigitToString(digit) + "\"";
-        resp                   = channel->cmd(command.getCmd() + dtmfString);
+        resp                   = channel->cmd(command.getCmd() + code.getDTMFCodeString());
         if (resp) {
             command = at::factory(at::AT::VTS);
-            resp    = channel->cmd(command.getCmd() + dtmfString);
+            resp    = channel->cmd(command.getCmd() + code.getDTMFCodeString());
```

Comment:
I believe that with `operator str()` this line could be not changed :)

Reply:
sure, that's a nice idea